CorsixTH/Lua/hospital.lua

```diff
@@ -183,18 +183,24 @@ function Hospital:Hospital(world, avail_rooms, name)
   self.policies["stop_procedure"] = 1 -- Note that this is between 1 and 2 ( = 100% - 200%)
   self.policies["goto_staffroom"] = 0.6
   self.policies["grant_wage_increase"] = TheApp.config.grant_wage_increase
+
   -- Randomly select three insurance companies to use, only different by name right now.
-  -- The first ones are more likely to come
+  -- The companies in the first quarter of the list are more likely to be selected
   self.insurance = {}
+  -- Make a local writeable copy table of the translated company names.
+  local companies = {}
   for no, local_name in ipairs(_S.insurance_companies) do
-    -- NOTE: Will not work if more companies are added
-    if math.random(1, 11) < 4 or 11 - no < #self.insurance + 3 then
-      self.insurance[#self.insurance + 1] = local_name
-    end
-    if #self.insurance > 2 then
-      break
-    end
+    companies[no] = local_name
   end
+  repeat
+    local num = math.random(1, 2) == 1 and math.random(1, math.ceil(#companies / 4)) or
+        math.random(1, #companies)
+    if companies[num] then
+      self.insurance[#self.insurance + 1] = companies[num]
+      table.remove(companies, num)
+    end
+  until #self.insurance == 3 or #companies < 1
```

Comment:
Is there any consequence of using a repeat loop where #companies is actually <1? I wonder if a while loop may be better to protect against that. Actually if #companies is <1, line 197 will throw an error as arg 1 > arg 2.

Reply:
(That line above might only actually apply if we started with companies<1 ... if it naturally diminishes that's fine)